---
collection: teaching
title: "Project"
---

## Overview

## Deliverables

## Important Dates



## Grading

Overall, your project will be graded according to the following percentages:
* Final essay (50%). Your final essay will be graded based on both the quality
    of your scientific work and the written quality of the essay.
* Presentation (30%). Your presentation will also be judged based on scientific
    merit and your ability to present this scientific work intelligently and
    clearly to an audience of your peers who are unfamiliar with this work.
* Other deliverables (20%). To ensure you are completing the project in a
    timely fashion, we expect you to complete all other deliverables. Each
    of the five other deliverables will be worth 4% of your total score.

### Project Guidelines

Here are some guidelines for each component of the project to keep in mind.
This is not a rubric (i.e., we will not grade your project only according to
these guidelines), but things that will ensure that you are on track.

#### Essay: quality of writing

#### Essay: quality of scientific work

#### Presentation

## Project ideas

The project is deliberately open-ended; you may choose to complete a project on
any aspect of computational biology that you find interesting. However, we are
providing some example topics below. You may be interested in one of these, or
you may like to pick you own. It is acceptable if more than one group completes
the same project.

Furthermore, it is perfectly reasonable to begin one's project by examining
existing research. In fact, in some cases, replication of an existing paper may
be challenging enough to constitute a project.

* Compare the quality of the output of a family of different software programs
    for genome assembly on different types of organisms and read sets.
* Build a protein folding simulation that predicts the structure of a protein
    from its sequecne of amino acids, perhaps including motif information from
    DNA as well.
* Use genotyping data to identify the population structure of a species (e.g.,
    humans) and identify admixture in individuals.
* Classify a large family of cellular or medical images using deep learning
    with Tensorflow or PyTorch; in the case of medical images, extend this
    approach to provide an automated diagnosis of a patient based on their
    image(s).
* Build your own genome assembler that incorporates data from long reads or
    from "H-C" data finding chromatin contacts at different points of the
    genome.
* Replicate a metagenome assembler that can assemble multiple genomes from a
    single sample containing many species.
* Write an algorithm that will design primers for testing the presence of an
    arbitrary virus within an individual.
* Apply RNA sequencing to differentiate c ells taken from the same tissue in
    different organisms, or from different tissues in the same organism.
* Build and analyze a gene co-expression graph from RNA-sequencing data.
* Build a transcription factor network connecting transcription factors to the
    genes they regular, and infer what biological conclusions can be drawn from
    the properties of the resulting graph.
* Apply game theory to understand evolutionary dynamics.
* Construct a family of evolutionary trees (perhaps using multiple tree
    construction algorithms) for a variety of multiple alignments on the same
    collection of taxa to obtain a collection of "gene trees". Then, design an
    approach that reconciles these differing trees into a single "species tree"
    for the collection.
* Extend the evolutionary tree model to handle recombination/horizontal gene
    transfer (e.g., in influenza viruses).
* Find patterns across human microbiome data in five different tissues for 300
    different humans as part of the [Human Microbiome Project](https://www.hmpdacc.org/). Or, identify which bacteria (or lack thereof) may be implicated in certain diseases.
* Analyze [The Cancer Genome Atlas expression data](https://cancergenome.nih.gov) to find which expression patterns are associated with differing cancer types.
* Gather patient data for the spread of a virus, and compare it against
theoretical models.

## Additional resources

### Bioinformatics software resources

You may find the following lists of software resources useful:
* [De novo genome asselbmers](

### Bioinformatics research

You could check out the following venues for recently published research in
bioinformatics. Papers should be available through conference websites or on
preprint servers such as [bioRxiv](https://www.biorxiv.org/) or [arXiv](https://arxiv.org/).
* [RECOMB](https://recomb.org/recomb2024/program.html)
* [Workshop on Algorithms in Bioinformatics](https://algo-conference.org/2024/wabi/)

### Writing and presenting

UM has a [writing center](https://www.umt.edu/writing-center/default.php)
where you can get help with the written and
presentation components of the project if needed.

### Scientific merit

If your group is struggling with the scientific component of the project,
please reach out to the instructor for help.
